docs/Secure-Coding-Guide-for-Python/CWE-703/CWE-755/README.md

```diff
@@ -0,0 +1,194 @@
+# CWE-755: Improper Handling of Exceptional Conditions
+
+There are two built-in Python modules that allow for file manipulation. The original module is called `os`. It provides the ability to perform various functions related to the operating system, among which are creating, updating, and deleting files [[Python docs - os]](https://docs.python.org/3/library/os.html). Python 3.4 introduced the `pathlib`  module, which represents file paths as objects as opposed to string variables [[Python docs - pathlib]](https://docs.python.org/3/library/pathlib.html).
+
+Both modules use the same OS exceptions for file-related errors. Here are some of the common file-related exceptions:
+
+|Exception name|Description|
+|:----|:----|
+|`OSError`|Base exception for all OS-related error.|
+|`FileExistsError`|Raised when trying to create a file or directory which already exists.|
+|`FileNotFoundError`| Raised when a file or directory is requested but doesn’t exist.|
+|`IsADirectoryError`|Raised when a file operation is requested on a directory.|
+|`NotADirectoryError`|Raised when a directory operation is requested on something which is not a directory.|
+|`PermissionError`|Raised when trying to run an operation without adequate access rights - for example, filesystem permissions.|
+
+The full list of OS exceptions can be found in the Python documentation [[Python docs - OS Exceptions](https://docs.python.org/3/library/exceptions.html#os-exceptions)].
+
+It is important to handle those exceptions when performing file I/O operations.
+
+## Non-Compliant Code Example (os.remove()/os.unlink())
+
+This code example demonstrates an attempt to read a non-existent file using the os module. The `read_file()` function opens a file and reads its content using `os.open()` and `os.read()`. If the file does not exist, an `OSError`  or `FileNotFoundError` will be raised when trying to access the randomly generated file name.
+
+*[noncompliant01.py](noncompliant01.py):*
+
+```py
+""" Non-compliant Code Example """
+ 
+import os
+import uuid
+ 
+ 
+def read_file(file):
+    """Function for opening a file and reading it's content."""
+    fd = os.open(file, os.O_RDONLY)
+    content = os.read(fd)
+    return content.decode()
+ 
+ 
+#####################
+# exploiting above code example
+#####################
+# Attempting to read a random non-existent file
+read_file(f"{uuid.uuid4()}.txt")
+```
+
+## Compliant Solution (try/except blocks)
```

Comment:
The (try/except blocks) part was not there originally. Personally, I prefer it this way. Just add backticks to make it consistent with my previous comment:

## Compliant Solution (`try/except` blocks)